homeassistant/components/bmw_connected_drive/device_tracker.py

```diff
@@ -52,39 +47,39 @@ class BMWDeviceTracker(BMWConnectedDriveBaseEntity, TrackerEntity):
 
     def __init__(
         self,
-        account: BMWConnectedDriveAccount,
+        coordinator: BMWDataUpdateCoordinator,
         vehicle: ConnectedDriveVehicle,
     ) -> None:
         """Initialize the Tracker."""
-        super().__init__(account, vehicle)
+        super().__init__(coordinator, vehicle)
 
         self._attr_unique_id = vehicle.vin
-        self._location = pos if (pos := vehicle.status.gps_position) else None
         self._attr_name = vehicle.name
 
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Return entity specific state attributes."""
+        return {ATTR_DIRECTION: self.vehicle.status.gps_heading}
+
     @property
     def latitude(self) -> float | None:
         """Return latitude value of the device."""
-        return self._location[0] if self._location else None
+        return (
+            self.vehicle.status.gps_position[0]
+            if self.vehicle.is_vehicle_tracking_enabled
+            else None
+        )
 
     @property
     def longitude(self) -> float | None:
         """Return longitude value of the device."""
-        return self._location[1] if self._location else None
+        return (
+            self.vehicle.status.gps_position[1]
+            if self.vehicle.is_vehicle_tracking_enabled
+            else None
+        )
 
     @property
     def source_type(self) -> Literal["gps"]:
         """Return the source type, eg gps or router, of the device."""
         return SOURCE_TYPE_GPS
-
-    def update(self) -> None:
```

Comment:
Override `def _handle_coordinator_update(self) -> None:` instead so you don't have to convert back to using the long hand `@property` method

Reply:
Adjusted.